cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -411,6 +411,12 @@ public void onNamespaceCreation(String cdapNamespace, Map<String, String> proper
       throw new IOException(String.format("Cannot create Kubernetes namespace for %s because no name was provided",
                                           cdapNamespace));
     }
+    // Kubernetes namespace must be a lowercase RFC 1123 label, consisting of lower case alphanumeric characters or '-'
+    // and must start and end with an alphanumeric character
+    Pattern kubernetesRegex = Pattern.compile("[a-z0-9]([-a-z0-9]*[a-z0-9])?");
```

Comment:
Instead of creating the pattern here, define it as a private static final member similar to LABEL_PATTERN, NAMESPACE_LABEL_PATTERN

Reply:
done